### PR TITLE
[TEVA-2568] Automatically highlight gaps in employment

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-2568

## Screenshots of UI changes:
Screenshot 2021-07-26 at 15 56 29
Screenshot 2021-07-26 at 15 56 39

### CODE REVIEW

File: app/controllers/jobseekers/job_applications/breaks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    redirect_to back_path, success: t(".success")
+  end
+
+  private
+
+  def back_path
+    @back_path ||= jobseekers_job_application_build_path(job_application, :employment_history)
+  end
+
+  def employment
```
Comment (josephhull676) [inline app/controllers/jobseekers/job_applications/breaks_controller.rb:33]: employment_break?

File: app/controllers/jobseekers/job_applications/breaks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  def back_path
+    @back_path ||= jobseekers_job_application_build_path(job_application, :employment_history)
+  end
+
+  def employment_break
+    job_application.employments.break.find(params[:id] || params[:break_id])
+  end
+
+  def employment_params
```
Comment (josephhull676) [inline app/controllers/jobseekers/job_applications/breaks_controller.rb:37]: If the above is `employment_break`, should this be `employment_break_params `?

File: app/views/jobseekers/job_applications/build/qualifications.html.slim [hunk 3]
```diff
@@ -33,8 +33,8 @@
                                     tag.div(qualification.finished_studying_details.presence, class: "govuk-body")]))
 
           - detail.actions do
-            = govuk_button_to t("buttons.delete"), jobseekers_job_application_qualification_path(job_application, qualification), method: :delete, class: "govuk-delete-link govuk-!-margin-bottom-0", form_class: "inline-block button_to"
-            = govuk_link_to t("buttons.change"), edit_jobseekers_job_application_qualification_path(job_application, qualification), class: "govuk-link--no-visited-state inline-block"
+            = govuk_link_to t("buttons.change"), edit_jobseekers_job_application_qualification_path(job_application, qualification), class: "govuk-link--no-visited-state"
+            = govuk_link_to t("buttons.delete"), jobseekers_job_application_qualification_path(job_application, qualification), method: :delete
```
Comment (josephhull676) [inline app/views/jobseekers/job_applications/build/qualifications.html.slim:37]: Why are you changing this in this PR?

File: app/views/jobseekers/job_applications/build/references.html.slim [hunk 4]
```diff
@@ -23,8 +23,8 @@
                 - c.slot(:row, key: t("jobseekers.job_applications.references.#{attribute}"), value: reference[attribute])
 
           - detail.actions do
-            = govuk_button_to t("buttons.delete"), jobseekers_job_application_reference_path(job_application, reference), method: :delete, class: "govuk-delete-link govuk-!-margin-bottom-0", form_class: "inline-block button_to"
-            = govuk_link_to t("buttons.change"), edit_jobseekers_job_application_reference_path(job_application, reference), class: "govuk-link--no-visited-state inline-block"
+            = govuk_link_to t("buttons.change"), edit_jobseekers_job_application_reference_path(job_application, reference), class: "govuk-link--no-visited-state"
+            = govuk_link_to t("buttons.delete"), jobseekers_job_application_reference_path(job_application, reference), method: :delete
```
Comment (josephhull676) [inline app/views/jobseekers/job_applications/build/references.html.slim:27]: Same here?

File: app/views/shared/job_application/_explained_employment_break.html.slim [hunk 5]
```diff
@@ -0,0 +1,9 @@
+- accordion.add_section title: "", html_attributes: { class: "govuk-accordion__section--expanded employment-break", id: "employment-break-#{index}" } do
+  = govuk_inset_text classes: "govuk-!-margin-top-3 govuk-!-margin-bottom-3" do
+    h2.govuk-heading-s class="govuk-!-margin-bottom-1" = t("jobseekers.job_applications.build.employment_history.break")
+    p.govuk-hint class="govuk-!-margin-top-1 govuk-!-margin-bottom-1" #{employment.started_on} to #{employment.ended_on || "present"}
+
+  - unless employment.reason_for_break.empty?
+    .employment-break__reason class="govuk-!-margin-bottom-5"
+      h2.govuk-heading-s = "Break in work history details"
```
Comment (josephhull676) [inline app/views/shared/job_application/_explained_employment_break.html.slim:8]: Should this be a translation instead?

**Review by josephhull676 — APPROVED** [review]

File: app/views/jobseekers/job_applications/build/qualifications.html.slim (hunk 3 above)
Comment (cesidio) [reply app/views/jobseekers/job_applications/build/qualifications.html.slim:37]: to be consistent and have it in the right order and with the right style, it should not be a button

**Pull request merged**